src/main/routes/watch-route.ts

```diff
@@ -65,50 +90,72 @@ class ApiWatcher {
     this.watchRequest.abort();
   }
 
-  private sendEvent(evt: any) {
-    // convert to "text/event-stream" format
-    this.response.write(`data: ${JSON.stringify(evt)}\n\n`);
+  private sendEvent(evt: IKubeWatchEvent) {
+    this.response.write(`${JSON.stringify(evt)}\n`);
   }
 }
 
 class WatchRoute extends LensApi {
+  private response: ServerResponse;
+
+  private setResponse(response: ServerResponse) {
+    // clean up previous connection and stop all corresponding watch-api requests
+    // otherwise it happens only by request timeout or something else..
+    this.response?.destroy();
+    this.response = response;
+  }
 
-  public async routeWatch(request: LensApiRequest) {
-    const { response, cluster} = request;
-    const apis: string[] = request.query.getAll("api");
-    const watchers: ApiWatcher[] = [];
+  public async routeWatch(request: LensApiRequest<IWatchRoutePayload>) {
+    const { response, cluster, payload: { apis } = {} } = request;
 
-    if (!apis.length) {
+    if (!apis?.length) {
       this.respondJson(response, {
-        message: "Empty request. Query params 'api' are not provided.",
-        example: "?api=/api/v1/pods&api=/api/v1/nodes",
+        message: "watch apis list is empty"
       }, 400);
 
       return;
     }
 
-    response.setHeader("Content-Type", "text/event-stream");
+    this.setResponse(response);
+    response.setHeader("Content-Type", "application/json");
     response.setHeader("Cache-Control", "no-cache");
     response.setHeader("Connection", "keep-alive");
     logger.debug(`watch using kubeconfig:${JSON.stringify(cluster.getProxyKubeconfig(), null, 2)}`);
```

Comment:
Logger already has this functionality and logging should be single lines
```suggestion
    logger.debug("watch using kubeconfig", { kubeconfig: cluster.getProxyKubeconfig() });
```